src/usePlotly.js

```diff
@@ -0,0 +1,50 @@
+import { useLayoutEffect, useState, useMemo } from 'react';
+import { head, prop, compose, pick, objOf, mergeDeepRight } from 'ramda';
+import { stream, scan } from 'flyd';
+
+/**
+* A simple debouncing function
+*/
+const debounce = (fn, delay) => {
+   let timeout;
+
+   return function (...args) {
+      const functionCall = () => fn.apply(this, args);
+
+      timeout && clearTimeout(timeout);
+      timeout = setTimeout(functionCall, delay);
+   };
+};
+
+const getSizeForLayout = compose(objOf('layout'), pick(['width', 'height']), prop('contentRect'), head);
+
+export default function usePlotly() {
+   const updates = useMemo(stream, []);
+   const appendData = useMemo(stream, []);
+   const plotlyState = useMemo(
+      () => scan(mergeDeepRight, { data: [], config: {}, layout: {} }, updates),
+      []
```

Comment:
Shouldn't `updates` be added to the dependencies list here since it's used inside the hook?

Reply:
since `updates` is memoized with an empty array, it will only be computed once on mount, so adding it to the dependencies array or not for `plotlyState` will have no effect....